File: app/src/main/java/com/gas/app/ui/fragment/mine/MineFragment.java
package com.gas.app.ui.fragment.mine;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.base.lib.base.LazyLoadFragment;
import com.base.lib.di.component.AppComponent;

import com.gas.app.ui.fragment.mine.mvp.MineContract;
import com.gas.app.ui.fragment.mine.mvp.MinePresenter;
import com.gas.app.R;
import com.gas.app.ui.fragment.mine.di.DaggerMineComponent;

/**
 * Author:
 * Time: 2016/8/12 14:28
 */
public class MineFragment extends LazyLoadFragment<MinePresenter> implements MineContract.View {

    public static final String TAG = "MineFragment";





    @Override
    public void setupFragmentComponent(@NonNull AppComponent appComponent) {
        DaggerMineComponent.builder().appComponent(appComponent).view(this).build().inject(this);
    }

    @Override
    public View initView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate( R.layout.fragment_mine,container,false);
    }





    @Override
    public void initData(@Nullable Bundle savedInstanceState) {


    }

    @Override
    public void setData(@Nullable Object data) {

    }


    public static MineFragment newInstance() {
        return new MineFragment();
    }


    @Override
    public Context getWrapContext() {
        return mContext;
    }



    @Override
    protected void lazyLoadData() {

    }


    @Override
    public void showMessage(@NonNull String message) {

    }
}

File: module_zhihu/src/main/java/com/gas/zhihu/ui/fragment/mine/MineFragment.kt
package com.gas.zhihu.ui.fragment.mine

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.base.lib.base.BaseFragment
import com.base.lib.di.component.AppComponent
import com.gas.zhihu.R
import com.gas.zhihu.ui.fragment.mine.di.DaggerMineComponent
import com.gas.zhihu.ui.fragment.mine.mvp.MineContract
import com.gas.zhihu.ui.fragment.mine.mvp.MinePresenter


class MineFragment : BaseFragment<MinePresenter?>(), MineContract.View {
    override fun setupFragmentComponent(appComponent: AppComponent) {
        DaggerMineComponent.builder().appComponent(appComponent)!!.view(this)!!.build()!!.inject(this)
    }

    override fun initView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.zhihu_fragment_mine, container, false)
    }

    override fun initData(savedInstanceState: Bundle?) {}
    override fun setData(data: Any?) {}
    override val wrapContext: Context
        get() = mContext

    override fun showMessage(message: String) {}

    companion object {
        const val TAG = "MineFragment"
        fun newInstance(): MineFragment {
            return MineFragment()
        }
    }
}